Code-the-Dream-School/ee-prac-team3-front
# Feature/user progress

## Purpose:
To allow user to see their progress on individual quizzes as they progress in their studies.

## Summary:
Updated front end to:
-removed previous standalone quiz progress data
-updated quiz schema to accommodate user related data regarding quiz progress
-Adds useEffect hooks to update quiz user data on the Main page and after completing a quiz on the QuizContent component
-Updates quiz context on completion of quiz to trigger re render of quiz page (might be better just to trigger a reload of the page when navigating back?)
- tracks highest score and number of attempts on quiz card for each quiz

## Front End Change:
App.js
Main.jsx
QuizContent.jsx
QuestionContent.jsx
QuizContextProvider.jsx
exportFunctions.js

## Test Steps:
Reviewer must be running on the userProgress branch of the backend.  User can log in with any credentials and should be able to take a quiz and see progress and attempt count change after submitting the test.
